devel/grpc/Portfile

```diff
@@ -1,53 +1,114 @@
+                    anywhere. gRPC enables client and server \
+                    applications to communicate transparently, and \
+                    simplifies the building of connected systems.
 
-checksums       rmd160  d1a5675ca59564385e93cbadabd159b508a9be99 \
-                sha256  812ab0caa6474b8a7efcaa77f48f5f35c865ab72fa8eac3ed52d050733cf4e8e \
-                size    15081393
+homepage            https://grpc.io
+platforms           darwin
 
-compiler.cxx_standard   2011
+checksums           rmd160  8e0913d0703be57b9d65aeb45db4aa733a292fff \
+                    sha256  7322710cae7ea2e11f268016f229b308e09854f12345dfe9e8638488609d8a66 \
+                    size    6979857
 
-patchfiles  patch-Makefile.diff \
-            grpc_if_nametoindex_posix.cc.diff
+# ignore pre releases
+github.livecheck.regex  {([0-9.]+)}
+
+compiler.cxx_standard   2011
 
 depends_lib-append \
-    port:zlib \
-    port:protobuf3-cpp \
-    port:c-ares \
-    path:lib/libssl.dylib:openssl
+                    path:lib/libssl.dylib:openssl \
+                    port:abseil \
+                    port:c-ares \
+                    port:gflags \
+                    port:gperftools \
+                    port:lbzip2 \
+                    port:libunwind \
+                    port:protobuf3-cpp \
+                    port:re2 \
+                    port:zlib
+
+depends_build-append \
+                    port:cctools \
+                    port:ld64 \
+                    port:pkgconfig
+
+post-fetch {
+    system -W ${worksrcpath}/third_party "git submodule update --init benchmark"
+    system -W ${worksrcpath}/third_party "git submodule update --init bloaty"
+    system -W ${worksrcpath}/third_party "git submodule update --init envoy-api"
+    system -W ${worksrcpath}/third_party "git submodule update --init gflags"
+    system -W ${worksrcpath}/third_party "git submodule update --init googleapis"
+    system -W ${worksrcpath}/third_party "git submodule update --init googletest"
+    system -W ${worksrcpath}/third_party "git submodule update --init libuv"
+    system -W ${worksrcpath}/third_party "git submodule update --init protoc-gen-validate"
+    system -W ${worksrcpath}/third_party "git submodule update --init udpa"
+}
+
+configure.args-append \
+                    -DBUILD_SHARED_LIBS=ON \
+                    -DCMAKE_CXX_STANDARD=11 \
+                    -DgRPC_ABSL_PROVIDER=package \
+                    -DgRPC_CARES_PROVIDER=package \
+                    -DgRPC_PROTOBUF_PROVIDER=package \
+                    -DgRPC_RE2_PROVIDER=package \
+                    -DgRPC_SSL_PROVIDER=package \
+                    -DgRPC_ZLIB_PROVIDER=package
+
+# the build must link against its own libraries
+configure.pre_args-replace \
+                    -DCMAKE_INSTALL_NAME_DIR="${cmake.install_prefix}/lib" \
+                    -DCMAKE_INSTALL_NAME_DIR="${cmake.build_dir}"
+
+build.target
 
-depends_build-append port:pkgconfig
+# fix shared library paths
```

Comment:
What's wrong with the shared library paths, and could the build system be fixed to generate them correctly in the first place, rather than adding all this difficult-to-understand code to the portfile?

Reply:
Here's the issue:

The build links against its own shared libraries in `${cmake.build_dir}`. But these these libraries are built with id `${prefix}/lib`. Because they haven't been installed yet, this causes an "image not found" linker error during the build.

If there's a smarter way to tackle this, I'm not aware of it.

Reply:
check this out <https://github.com/macports/macports-ports/blob/d7d881884bbb082a572bb24ddaebcd41fe553e24/_resources/port1.0/group/cmake-1.1.tcl#L105>

Reply:
That's the smarter way—thanks!